pallets/xst/benchmarking/src/lib.rs

```diff
@@ -0,0 +1,272 @@
+        )
+    }
+}
+pub struct Pallet<T: Config>(xst::Pallet<T>);
+pub trait Config: xst::Config + band::Config + oracle_proxy::Config + price_tools::Config {}
+
+benchmarks! {
+    set_reference_asset {
+    }: _(
+        RawOrigin::Root,
+        DAI.into()
+    )
+    verify {
+        utils::assert_last_event::<T>(Event::ReferenceAssetChanged(DAI.into()).into())
+    }
+
+    enable_synthetic_asset {
+        utils::relay_symbol::<T>()?;
+    }: _(
+        RawOrigin::Root,
+        XSTUSD.into(),
+        utils::symbol(),
+        fixed!(0)
+    )
+    verify {
+        assert!(
+            XSTPool::<T>::enabled_symbols(
+                utils::symbol::<<T as xst::Config>::Symbol>()
+            )
+            .is_some()
+        );
+    }
+
+    disable_synthetic_asset {
+        let asset_id = utils::enable_synthetic_asset::<T>()?;
+    }: _(
+        RawOrigin::Root,
+        asset_id
+    )
+    verify {
+        utils::assert_last_event::<T>(Event::SyntheticAssetDisabled(asset_id).into())
+    }
+
+    register_synthetic_asset {
+        let permissioned_account_id = utils::permissioned_account_id::<T>();
+        let reference_symbol = utils::symbol::<<T as xst::Config>::Symbol>();
+    }: _(
+        RawOrigin::Root,
+        AssetSymbol(b"XSTEURO".to_vec()),
+        AssetName(b"Sora Synthetic EURO".to_vec()),
+        reference_symbol
+    )
+    verify {
+        utils::assert_last_assets_event::<T>(
+            AssetsEvent::AssetRegistered(
+                utils::symbol_asset_id::<T>(),
+                permissioned_account_id
+            ).into()
+        );
+    }
+
+    set_synthetic_asset_fee {
+        let asset_id = utils::enable_synthetic_asset::<T>()?;
+        let fee_ratio = fixed!(0.06);
+    }: _(
+        RawOrigin::Root,
+        asset_id.clone(),
+        fee_ratio
+    )
+    verify {
+        utils::assert_last_event::<T>(Event::SyntheticAssetFeeChanged(asset_id, fee_ratio).into())
+    }
+
+    set_synthetic_base_asset_floor_price {
+    }: _(RawOrigin::Root, balance!(200))
+    verify {
+        utils::assert_last_event::<T>(Event::SyntheticBaseAssetFloorPriceChanged(balance!(200)).into())
+    }
+
+    quote {
+        utils::setup_exchange_benchmark::<T>();
+        let asset_id = utils::enable_synthetic_asset::<T>()?;
+    }: {
+        let _ = XSTPool::<T>::quote(
+            &DEXId::Polkaswap.into(),
+            &XST.into(),
+            &asset_id,
+            QuoteAmount::with_desired_input(balance!(1)),
+            true,
+        ).unwrap();
+    }
+
+    exchange {
+        utils::setup_exchange_benchmark::<T>();
+        let asset_id = utils::enable_synthetic_asset::<T>()?;
+    }: {
+        let _ = XSTPool::<T>::exchange(
+            &utils::alice::<T>(),
+            &utils::alice::<T>(),
+            &DEXId::Polkaswap.into(),
+            &XST.into(),
+            &asset_id,
+            SwapAmount::with_desired_input(balance!(1), 1),
+        ).unwrap();
+    }
```

Comment:
Did you test it as unit tests and as benchmarks as well?

Reply:
Yes, both cases